content/socials.js

```diff
@@ -67,23 +74,44 @@ export default class Socials {
     const matcher = /(?:https?:\/\/)?(?:www.)?facebook.(?:[a-z]+)\/([^/?\s]+)/i;
 
     return {
-      matcher
+      match: (input) => {
+        if (!matcher.test(input)) {
+          return null;
+        }
+
+        const matches = matcher.exec(input);
+        return matches[1];
+      }
     };
   }
 
   static get twitter () {
     const matcher = /(?:https?:\/\/)?(?:www.)?twitter.(?:[a-z]+)\/([^/?\s]+)/i;
 
     return {
-      matcher
+      match: (input) => {
+        if (!matcher.test(input)) {
+          return null;
+        }
+
+        const matches = matcher.exec(input);
+        return matches[1];
+      }
     };
   }
 
   static get reddit () {
     const matcher = /(?:https?:\/\/)?(?:www.)?reddit.(?:[a-z]+)\/u\/([^/?\s]+)/i;
 
     return {
-      matcher
+      match: (input) => {
+        if (!matcher.test(input)) {
+          return null;
+        }
+
+        const matches = matcher.exec(input);
```

Comment:
Just out of curiosity, I saw this pattern in couple of places, what's the advantage of using `.test` and then `.exec`?

Wouldn't that work:
```js
const matches = matcher.exec(input);
return matches && matches[1];
```